Transactions List - indicate when a transaction is expected to be included in a future deposit

Jinksi opened the pull request:
Fixes #7876

#### Changes proposed in this Pull Request

In the Transactions List, for transactions that are not included in a pending/paid deposit, render "Future deposit" and an info icon in the "Deposit date" column.

Clicking this icon will present a tooltip with the message:

```
This transaction will be included in a future deposit. Once created, the estimated deposit date will be shown here.
```

This will render for all transaction types except `financing_*`. [See list of transaction types](https://github.com/Automattic/woocommerce-payments/blob/ceda996a805c0f556b55dbd10b665682f184112f/client/transactions/strings.ts#L9-L24).


<img width="1179" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/bbec8464-d56c-4f08-9a45-1983f6437dcf">


#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* View the Transactions List
* Ensure non-financing transactions that are not included in previous deposits show the tooltip icon.
* Clicking on the tooltip icon should show the message defined above.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Jinksi committed:
Tranasaction list: indicate when part of future deposit

Jinksi committed:
Tranasaction list: indicate when part of future deposit

Jinksi committed:
Add changelog entry

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7878` or branch name `add/transaction-list-future-deposit` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/transaction-list-future-deposit) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6b97f480d14c209d5e6e3524e91e5b6550d57ad7
- Build time: 2023-12-13 05:28:39 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Remove redundant estimated deposit test

Jinksi committed:
Update snapshot tests

Jinksi committed:
Don't show "Future deposit" for financing transactions

Jinksi committed:
Show tooltip message rather than "Future deposit"

shendy-a8c commented:
> This will render for all transaction types except financing_*.

Why is this? I'm not too familiar with loan feature. When merchant gets a loan, isn't the money paid out in the form of a deposit?

shendy-a8c reviewed (approved):
Looks good and tested well. I just have a question above, about loan transaction.

csmcneill commented:
In case merchants want more details, should we include a `Learn more.` link in the tooltip that directs users to https://woo.com/document/woopayments/deposits/deposit-schedule/#pending-funds ?

Jinksi commented:
> In case merchants want more details, should we include a Learn more. link in the tooltip that directs users to [woo.com/document/woopayments/deposits/deposit-schedule/#pending-funds](https://woo.com/document/woopayments/deposits/deposit-schedule/#pending-funds) ?

Great idea, I'll add this!

github-actions[bot] committed:
Update version and add changelog entries for release 6.9.2

Jinksi commented:
> When merchant gets a loan, isn't the money paid out in the form of a deposit? - @shendy-a8c 

At the moment, we show loan financing transactions as individual transactions separate from (rather than within) a deposit.

@souravdebnath1986 has mentioned that we want to rectify this in the future, at which point financing types will have a deposit date and will show up within the deposit details transaction list.

Example: below is the same deposit, including a financing repayment.

Stripe shows the financing repayment within the transactions list.

<img width="1415" alt="image 2" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/1c549632-d8af-44eb-b6fb-ba83e326c3b0">


WooPayments Deposit details transaction list doesn’t include this and therefore the totals don’t add up.
<img width="1299" alt="image 3" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/dccda093-46e2-4c13-89e0-ee94d2606cf9">


In 6.8.0, it is especially confusing, since these show an estimated deposit that never arrives.
<img width="1225" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/b645e14a-4b2e-4f4e-a3ee-1900e2ddb3bd">


I'll check to see if there is an open issue for this and create one if not.

Jinksi committed:
Merge branch 'release/6.9.2' into add/transaction-list-future-deposit

Jinksi commented:
Rebased to `release/6.9.2` and removed all commits carried over from `develop`.

Jinksi committed:
Reorder const declarations to keep deposits together

Jinksi committed:
Add learn more link to tooltip content

Jinksi committed:
Update tooltip content wording

Jinksi committed:
Add "Future deposit" string alongside tooltip icon

Jinksi committed:
Merge branch 'release/6.9.2' into add/transaction-list-future-deposit

Jinksi committed:
Reorder imports

Jinksi commented:
I've updated this PR by adding a Learn More link that directs to `https://woo.com/document/woopayments/deposits/deposit-schedule/#pending-funds`.

I've also added the text "Future deposit" to the row's value, alongside the tooltip icon.

shendy-a8c commented:
Retested and looks good.
